src/llfs/ioring_log_device2.ipp

```diff
@@ -350,20 +383,37 @@ inline void IoRingLogDriver2<StorageT>::start_flush(CommitPos observed_commit_po
       }
     }
 
+    // If burst mode optimization is enabled, then round slot_range.upper_bound *down* to get the
+    // aligned upper bound (instead of up, the default); this way, we are less likely to need to
+    // issue another I/O to fill in partial data in flush_tail_.
+    //
+    if (this->flush_tail_ && this->options_.optimize_burst_mode) {
+      this->metrics_.burst_mode_checked.add(1);
+
+      const usize new_upper_bound = slot_max(
+          slot_range.lower_bound,
+          batt::round_down_bits(this->config_.data_alignment_log2, slot_range.upper_bound));
+
+      if (slot_range.upper_bound != new_upper_bound) {
+        this->metrics_.burst_mode_applied.add(1);
+      }
+
+      slot_range.upper_bound = new_upper_bound;
+    }
```

Comment:
Do you think it will be simpler to get ride of flush_tail handling completely? In that case flush will not cover partial page (the end page if its not fully written).  This way subsequent flush will take care of this last flush's unwritten page without any special handling. This will simplify the code in start_flush and handle_flush_write routines. With this basically optimized_flush will be always ON.

Reply:
This will also make sure there is no overlapping page transfers, thus we will not need any handling for that.

Reply:
So, if I understand correctly, the idea would be to not do any concurrent writing if the highest-offset flush write isn't aligned to a page boundary?  I think you're right that this could be done without any `flush_tail_` member data at all.  It's an interesting idea... it wouldn't avoid the need to overwrite the non-aligned page, but it would make the code simpler since that overwriting would never be concurrent to some other write.  Let me see if I can collect some data to see how this change would impact performance.

Reply:
That's right. My thinking is that a flush will never cover partial page (could be also called as the non-aligned page which is the end page). This partial page will be covered by the next (future) flush for which it will be its first fully filled page. There is no overwrite here.